backend/dal/Services/Admin/Concrete/ParcelService.cs

```diff
@@ -364,8 +364,20 @@ public override void Update(Parcel parcel)
             parcel.ThrowIfNull(nameof(parcel));
             parcel.ThrowIfNotAllowedToEdit(nameof(parcel), this.User, new[] { Permissions.SystemAdmin, Permissions.AgencyAdmin });
 
-            var originalParcel = this.Context.Parcels.Find(parcel.Id) ?? throw new KeyNotFoundException();
+            var originalParcel = this.Context.Parcels
+                .Include(p => p.Agency)
+                .Include(p => p.Address)
+                .Include(p => p.Evaluations)
+                .Include(p => p.Fiscals)
+                .Include(p => p.Buildings).ThenInclude(pb => pb.Building).ThenInclude(b => b.Evaluations)
+                .Include(p => p.Buildings).ThenInclude(pb => pb.Building).ThenInclude(b => b.Fiscals)
+                .Include(p => p.Buildings).ThenInclude(pb => pb.Building).ThenInclude(b => b.Address)
+                .Include(p => p.Parcels).ThenInclude(pp => pp.Parcel)
+                .Include(p => p.Subdivisions).ThenInclude(pp => pp.Subdivision)
+                .SingleOrDefault(p => p.Id == parcel.Id) ?? throw new KeyNotFoundException();
+
             parcel.PropertyTypeId = originalParcel.PropertyTypeId;
+            this.Context.Entry(originalParcel.Address).CurrentValues.SetValues(parcel.Address);
```

Comment:
@LawrenceLau2020  Do we need to do something similar for the buildings as well?

Reply:
I think that depends on the requirements of this feature, I'm not sure if SRES needs to be able to bulk upload or bulk update buildings? I recall Lindsay saying it wasn't needed, but maybe we need to double check this.